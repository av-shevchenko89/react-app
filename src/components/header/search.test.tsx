import React from 'react';
import renderer from 'react-test-renderer';
import {render, fireEvent, waitFor, screen} from '@testing-library/react'
import {Search} from './search';

const mockFn = jest.fn();

it('renders correctly', () => {
  const tree = renderer
    .create(<Search searchMovie={mockFn}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it(' search value', async () => {
    render(<Search searchMovie={mockFn}/>)
  
    const value = 'search query';

    fireEvent.change(screen.getByRole('textbox'), {target: {value}});
    fireEvent.click(screen.getByRole('button'));

    expect(mockFn).toHaveBeenCalledWith(value);
  })