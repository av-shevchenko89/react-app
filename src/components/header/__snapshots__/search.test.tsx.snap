// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="search"
>
  <input
    className="form-control"
    onChange={[Function]}
    placeholder="What do you want to watch?"
    type="text"
    value=""
  />
  <button
    className="btn"
    onClick={[Function]}
  >
    Search
  </button>
</div>
`;
